lib/Service/CustomerService.php

```diff
@@ -0,0 +1,267 @@
+     * @return \Stripe\CustomerBalanceTransaction
+     */
+    public function createBalanceTransaction($customerId, $params = [], $opts = [])
+    {
+        return $this->createNestedObject(self::BALANCE_TRANSACTIONS, $customerId, $params, $opts);
+    }
+
+    /**
+     * Create a source.
+     *
+     * @param string $customerId
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Account|\Stripe\AlipayAccount|\Stripe\BankAccount|\Stripe\BitcoinReceiver|\Stripe\Card|\Stripe\Source
+     */
+    public function createSource($customerId, $params = [], $opts = [])
+    {
+        return $this->createNestedObject(self::SOURCES, $customerId, $params, $opts);
+    }
+
+    /**
+     * Create a tax ID.
+     *
+     * @param string $customerId
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\TaxId
+     */
+    public function createTaxId($customerId, $params = [], $opts = [])
+    {
+        return $this->createNestedObject(self::TAX_IDS, $customerId, $params, $opts);
+    }
+
+    /**
+     * Delete a customer.
+     *
+     * @param string $id
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Customer
+     */
+    public function delete($id, $params = [], $opts = [])
+    {
+        return $this->deleteObject($id, $params, $opts);
+    }
+
+    /**
+     * Delete a customer discount.
+     *
+     * @param string $id
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Discount
+     */
+    public function deleteDiscount($id, $params = [], $opts = [])
+    {
+        return $this->request("delete", $this->instancePath($id) . "/discount", $params, $opts);
+    }
+
+    /**
+     * Delete a source.
+     *
+     * @param string $customerId
+     * @param string $id
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Account|\Stripe\AlipayAccount|\Stripe\BankAccount|\Stripe\BitcoinReceiver|\Stripe\Card|\Stripe\Source
+     */
+    public function deleteSource($customerId, $id, $params = [], $opts = [])
+    {
+        return $this->deleteNestedObject(self::SOURCES, $customerId, $id, $params, $opts);
+    }
+
+    /**
+     * Delete a tax ID.
+     *
+     * @param string $customerId
+     * @param string $id
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\TaxId
+     */
+    public function deleteTaxId($customerId, $id, $params = [], $opts = [])
+    {
+        return $this->deleteNestedObject(self::TAX_IDS, $customerId, $id, $params, $opts);
+    }
+
+    /**
+     * Retrieve a customer.
+     *
+     * @param string $id
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Customer
+     */
+    public function retrieve($id, $params = [], $opts = [])
+    {
+        return $this->retrieveObject($id, $params, $opts);
+    }
+
+    /**
+     * Retrieve a balance transaction.
+     *
+     * @param string $customerId
+     * @param string $id
```

Comment:
should the second parameter have the name `balanceTransactionId` to be clearer?

Reply:
Hm, maybe... Will keep as-is for now and add a note in the integration PR to decide if we want to change it.